src/game/Spells/Spell.cpp

```diff
@@ -792,7 +792,7 @@ void Spell::AddUnitTarget(Unit* target, uint8 effectMask, CheckException excepti
     targetInfo.heartbeatResistChance = 0;
 
     // Calculate hit result
-    targetInfo.missCondition = (m_ignoreHitResult ? SPELL_MISS_NONE : m_caster->SpellHitResult(target, m_spellInfo, targetInfo.effectMask, m_reflectable, &targetInfo.heartbeatResistChance));
+    targetInfo.missCondition = (m_ignoreHitResult ? SPELL_MISS_NONE : m_caster->SpellHitResult(target, m_spellInfo, targetInfo.effectMask, m_reflectable, &(targetInfo.heartbeatResistChance)));
```

Comment:
I'm not sure about this one, can somebody else confirm that this is correct?

Reply:
What's the warning message?

Reply:
```
field heartbeatResistChance
Type: uint32
Value = true

// In Spell::TargetInfo
public: uint32 heartbeatResistChance
Address of 'targetInfo.heartbeatResistChance' will always evaluate to 'true'clang(-Wpointer-bool-conversion)
```

Reply:
This one is an error. SpellHitResult prototype have probably changed over time and this happened. Will summon @killerwife for this. Ignore this for this pr.

Reply:
Fixed on my own accord.